operator/src/main/java/com/github/streamshub/console/dependents/ConfigurationProcessor.java

```diff
@@ -0,0 +1,744 @@
+                        for (var aliases = keystore.aliases().asIterator(); aliases.hasNext(); ) {
+                            alias = aliases.next();
+
+                            if (keystore.isCertificateEntry(alias)) {
+                                var certificate = keystore.getCertificate(alias);
+                                encodeCertificate(buffer, certificate);
+                            }
+                        }
+                    }
+                } catch (KeyStoreException | IOException | CertificateEncodingException e) {
+                    throw new ReconciliationException("Truststore %s could not be loaded. %s"
+                            .formatted(sourceName, e.getMessage()));
+                }
+
+                putTrustStoreValue(data, sourcePrefix + sourceName, "content.pem", buffer.toByteArray());
+
+                varsUI.add(new EnvVarBuilder()
+                        .withName("CONSOLE_SECURITY_OIDC_TRUSTSTORE")
+                        .withNewValueFrom()
+                            .withNewSecretKeyRef(sourcePrefix + sourceName + ".content.pem", secretName, false)
+                        .endValueFrom()
+                        .build());
+            } else if (uiServer) {
+                varsUI.add(new EnvVarBuilder()
+                        .withName("CONSOLE_SECURITY_OIDC_TRUSTSTORE")
+                        .withNewValueFrom()
+                            .withNewSecretKeyRef(sourcePrefix + sourceName + ".content", secretName, false)
+                        .endValueFrom()
+                        .build());
+            }
+        }
+
+        private boolean putTrustStoreValue(Map<String, String> data, String sourceName, String key, byte[] value) {
+            if (value != null) {
+                data.put(sourceName + "." + key, ConfigSupport.encodeBytes(value));
+                return true;
+            }
+            return false;
+        }
+
+        private void encodeCertificate(OutputStream buffer, Certificate certificate) throws IOException, CertificateEncodingException {
+            buffer.write("-----BEGIN CERTIFICATE-----\n".getBytes(StandardCharsets.UTF_8));
+            buffer.write(Base64.getMimeEncoder(80, new byte[] {'\n'}).encode(certificate.getEncoded()));
+            buffer.write("\n-----END CERTIFICATE-----\n".getBytes(StandardCharsets.UTF_8));
+        }
+    }
+
+    /**
+     * Generate additional entries in the secret for metric source trust stores. Also, this
+     * method will add to the context the resources to be added to the console deployment to
+     * access the secret entries.
+     */
+    private void buildTrustStores(Console primary, Context<Console> context, Map<String, String> data) {
+        var processor = new TrustStoreProcessor(primary, context, data);
+        processor.process();
+    }
+
+    private static String base64String(int length) {
+        ByteArrayOutputStream buffer = new ByteArrayOutputStream();
```

Comment:
nit - method name doesn't reflect the implementation.   (generateRandomSecret)
nit - use try with resources.